### DISCUSSION TITLE
FCM in China

### DISCUSSION THREAD

**Discussion by teckbeng-payboy** [invertase/react-native-firebase#5689] [Q&A]
HI, i have developed an app where it can be login and perform other function normally but when the user in China want to try out the app right, it will stuck at the login screen right after he submit his credential.

The code on the submission button is to get FCM token first. So I wonder is it FCM is not allow in China that's why my app keep loading? 

*Happen on both Android & IOS

**Comment by redoxide-dev** [comment]
Yes. you are right. I am not sure about all, but china block googleapis and google domain. but you are still good to go to use service supported by firebase domain. I still can use Firebase realtime database. The login stuff probably is using google and googleapis domain.

**Comment by teckbeng-payboy** [comment]
I actually not using the authentication by firebase, I just want to get a FCM token for notification purpose. Not sure is the FCM module related to the google and googleapis domain or not.. Will update here if I can get further details from the user..

**Reply by mikehardy** [reply]
You will likely want to check the upstream firebase-android-sdk / firebase-ios-sdk / firebase-js-sdk repos as well - those (and the upstream documentation) will probably offer more authoritative information on the subject, we won't be able to affect things like that here as we just wrap them
